crates/ra_lsp_server/src/req.rs

```diff
@@ -129,6 +129,7 @@ pub struct PublishDecorationsParams {
 pub struct Decoration {
     pub range: Range,
     pub tag: &'static str,
+    pub id: Option<String>,
```

Comment:
`id` seems like an unfortunate name.

Let's make it `binding_hash: Option<u64>`, with a short comment explaining how we use this for highlighting